mlx/backend/cuda/device.cpp

```diff
@@ -41,9 +52,12 @@ Device::Device(int device) : device_(device) {
   // The cublasLt handle is used by matmul.
   make_current();
   cublasLtCreate(&lt_);
+  // The cudnn handle is used by Convolution.
+  CHECK_CUDNN_ERROR(cudnnCreate(&cudnn_));
 }
 
 Device::~Device() {
+  CHECK_CUDNN_ERROR(cudnnDestroy(cudnn_));
   cublasLtDestroy(lt_);
```

Comment:
I guess we should also check cublas error when making and destroying the cublas handles right?

Reply:
Yeah we should do that, I updated the PR.